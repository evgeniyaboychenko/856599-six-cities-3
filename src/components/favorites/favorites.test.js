import React from 'react';
import renderer from 'react-test-renderer';
import Favorites from './favorites.jsx';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);
const CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
const COORDINATES_CITY = [[48.856663, 2.351556], [50.930779, 6.938399], [50.854283, 4.352131], [52.373057, 4.892557], [53.552645, 9.966287], [51.230569, 6.787428]];

jest.mock(`../offer-list/offer-list.jsx`);

const offersFavorite = [
  {
    city: {
      name: CITIES[0],
      coordinatesCity: COORDINATES_CITY[0],
      zoom: 10,
    },
    offersFavoriteByCity: [{
      id: 1,
      cityLocation: {
        coordinates: [48.856663, 2.351556], zoom: 10
      },
      cityName: CITIES[0],
      name: `Beautiful & luxurious studio at great location`,
      rating: 3,
      price: 100,
      type: `apartment`,
      isPremium: false,
      isFavorite: true,
      descriptions: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      photos: [`img/1.png`, `img/2.png`],
      previewImage: `img/1.png`,
      countRooms: 3,
      maxGuests: 4,
      appliances: [`Heating`, `Kitchen`],
      coordinates: [52.35514938496378, 4.673877537499948],
      zoom: 8,
      owner: {
        avatar: `img/1.png`, id: 3, name: `Angelina`, isSuper: true
      }
    },
    {
      id: 2,
      cityLocation: {
        coordinates: [48.856663, 2.351556], zoom: 10
      },
      cityName: CITIES[0],
      name: `Beautiful & luxurious studio at great location`,
      rating: 3,
      price: 100,
      type: `apartment`,
      isPremium: false,
      isFavorite: true,
      descriptions:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      photos: [`img/1.png`, `img/2.png`],
      previewImage: `img/1.png`,
      countRooms: 3,
      maxGuests: 4,
      appliances: [`Heating`, `Kitchen`],
      coordinates: [52.35514938496378, 4.673877537499948],
      zoom: 8,
      owner: {
        avatar: `img/1.png`, id: 3, name: `Angelina`, isSuper: true
      }
    }
    ]
  }
];

const offersFavoriteEmpty = [];
const user = {
  id: 1,
  name: `Nike`,
  avatar: `nike.jpg`,
  email: `nike@n.ru`,
  isPro: false,
};

it(`should Favorites render correctly`, () => {
  const store = mockStore({
  });
  const tree = renderer.create(
      <Provider store={store}>
        <Favorites
          offersFavorite = {offersFavorite}
          user = {user}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`should FavoritesEmpty render correctly`, () => {
  const store = mockStore({
  });
  const tree = renderer.create(
      <Provider store={store}>
        <Favorites
          offersFavorite = {offersFavoriteEmpty}
          user = {user}
        />
      </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
