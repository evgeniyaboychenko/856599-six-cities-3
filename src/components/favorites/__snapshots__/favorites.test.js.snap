// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Favorites render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "url(https://htmlacademy-react-3.appspot.com/six-citiesnike.jpg)",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  nike@n.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <li
            className="favorites__locations-items"
          >
            <div
              className="favorites__locations locations locations--current"
            >
              <div
                className="locations__item"
              >
                <a
                  className="locations__item-link"
                  href="#"
                >
                  <span>
                    Paris
                  </span>
                </a>
              </div>
            </div>
            <div
              className="favorites__places"
            />
          </li>
        </ul>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;

exports[`should FavoritesEmpty render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                  style={
                    Object {
                      "backgroundImage": "url(https://htmlacademy-react-3.appspot.com/six-citiesnike.jpg)",
                    }
                  }
                />
                <span
                  className="header__user-name user__name"
                >
                  nike@n.ru
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--favorites page__main--favorites-empty"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites favorites--empty"
      >
        <h1
          className="visually-hidden"
        >
          Favorites (empty)
        </h1>
        <div
          className="favorites__status-wrapper"
        >
          <b
            className="favorites__status"
          >
            Nothing yet saved.
          </b>
          <p
            className="favorites__status-description"
          >
            Save properties to narrow down search or plan yor future trips.
          </p>
        </div>
      </section>
    </div>
  </main>
  <footer
    className="footer container"
  >
    <a
      className="footer__logo-link"
      href="main.html"
    >
      <img
        alt="6 cities logo"
        className="footer__logo"
        height="33"
        src="img/logo.svg"
        width="64"
      />
    </a>
  </footer>
</div>
`;
